src/win32.c

```diff
@@ -1032,4 +1032,35 @@ gchar *win32_get_user_config_dir(void)
 	return g_build_filename(g_get_user_config_dir(), "geany", NULL);
 }
 
+
+/* Retrieve the console codepage
+ * In case GetConsoleOutputCP() returns 0 (i.e. the application doesn't have an own console window
+ * fallback to GetOEMCP(). */
+guint win32_get_console_codepage(void)
+{
+	guint codepage = GetConsoleOutputCP();
+	if (codepage == 0)
+		codepage = GetOEMCP();
+	return codepage;
+}
+
+
+/* Convert a string into the system's default codepage, this is different from the
+ * locale (e.g. default codepage is 850 but locale is CP1252).
+ * This assumes the input string is encoded as UTF-8, otherwise a copy of
+ * the input string is returned. */
+#define CODEPAGE_LEN 8
+gchar *win32_convert_to_system_codepage(const gchar *str, GError **error)
+{
+	if (g_utf8_validate(str, -1, NULL))
+	{
+		guint codepage_code = win32_get_console_codepage();
+		gchar codepage[CODEPAGE_LEN] = { 0 };
+		g_snprintf(codepage, CODEPAGE_LEN, "%u", codepage_code);
+		return g_convert(str, -1, codepage, "utf-8", NULL, NULL, error);
```

Comment:
https://gist.github.com/eht16/1305bf7cdb2d3504e62694e4e1141c40
This is `iconv -l | grep CP` and the list is too short to say "yes, everything is supported".
But I hope it is sufficient for most users.

Maybe there is some Windows API for converting between codepages. Not sure if it is worth the effort.